packages/docutils/lib/builder/nav.ts

```diff
@@ -0,0 +1,320 @@
+    .map((ent) => path.join(relDir, ent.name));
+}
+
+/**
+ * Because the `nav` property of `mkdocs.yml` is both a recursive type and a kind of awful one, it's
+ * easier to work with it if we rewrite the data into a flat array of objects. We keep a `keypath`
+ * prop which represents the deep/nested location within the `nav` object.
+ *
+ * @privateRemarks This function is not recursive; instead it loops over a queue of items to process
+ * data, and we append to that queue while processing if needed.
+ * @param nav Contents of the `nav` prop of `mkdocs.yml`
+ * @returns A list of objects, each with a `keypath` property and a `fileOrUrl` property (and maybe
+ * a `name` property)
+ */
+export function parseNav(nav: MkDocsYmlNav): ParsedNavData[] {
+  let parsedNav: ParsedNavData[] = [];
+  const entries = Object.entries(nav);
+  type QueueItem = {
+    entries: typeof entries;
+    keypath: string;
+  };
+  const queue: QueueItem[] = [{entries, keypath: ''}];
+
+  while (queue.length) {
+    const {entries, keypath} = queue.shift()!;
+    for (const [key, item] of entries) {
+      if (_.isString(item)) {
+        const navData: ParsedNavData = {
+          keypath: keypath ? `${keypath}.${key}` : key,
+          fileOrUrl: item,
+        };
+
+        // if the key is not convertible to a number, it's a name
+        // which was manually put there by somebody.
+        if (isNaN(Number(key))) {
+          navData.name = key;
+        }
+        parsedNav = [...parsedNav, navData];
+      } else if (_.isObject(item)) {
+        const subEntries = Object.entries(item);
+        queue.push({entries: subEntries, keypath: keypath ? `${keypath}.${key}` : key});
+      }
+    }
+  }
+  return parsedNav;
+}
+
+/**
+ * Finds all items within the list of parsed nav data which correpsond to the header.  This is
+ * imperfect, as it's possible for the header string to appear in multiple places in the nav, but let's just ignore that until we can't.
+ * @param navData Some parsed nav data
+ * @param header Header name
+ */
+function itemsForHeader(navData: ParsedNavData[], header: string) {
+  return _.filter(navData, (item) => _.toPath(item.keypath).includes(header));
+}
+
+/**
+ * Update the `nav` property of `mkdocs.yml` with a list of "command" files generated by TypeDoc via
+ * `@appium/typedoc-plugin-appium`.
+ *
+ * To be clear, this function **modifies the MkDocs config file (`mkdocs.yml`) in place**; it is
+ * typically under version control, so if this function makes any changes, you'll want to commit them.
+ * @param opts - Options
+ * @todo implement `dryRun` option
+ */
+export async function updateNav({
+  cwd = process.cwd(),
+  mkdocsYml: mkDocsYmlPath,
+  typedocJson: typeDocJsonPath,
+  all = false,
+  dryRun = false,
+}: UpdateNavOpts = {}) {
+  // we need `mkdocs.yml` to update
+  // and we need `typedoc.json` to know where to look for the command docs
+  [mkDocsYmlPath, typeDocJsonPath] = await Promise.all([
```

Comment:
I prefer to use B.all, it's more flexible to cases where the input size is zero

Reply:
that would mean introducing bluebird where I don't necessarily need it though.  the input size is known